crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -1833,6 +1783,17 @@ impl ScheduleGraph {
         graph: &DiGraph,
         report: ReportCycles,
     ) -> Result<Vec<NodeId>, ScheduleBuildError> {
+        // Check explicitly for self-edges.
+        // `iter_sccs` won't report them as cycles because they still form components of one node.
+        if let Some((node, _)) = graph.all_edges().find(|(left, right)| left == right) {
```

Comment:
Non-blocking, candidate for a follow up: We should report all self-edges rather than just the first one encountered.